File: readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

// proto typing
float countLetters(string text);
float countWords(string text);
float countSentences(string text);
void readerGrade(float index);

int main(void)
{
    string text = get_string("Text: ");
    // count the letters, count the words and count the sentences
    float nLetters = countLetters(text), nWords = countWords(text), nSentences = countSentences(text) || 1;

    // calculate the averages
    float L = (nLetters / nWords) * 100;
    float S = (nSentences / nWords) * 100;

    // The Coleman-Liau index
    float index = 0.0588 * L - 0.296 * S - 15.8;

    // print out the reading grade to the screen
    readerGrade(index);
}

// count the number of letters in a given text
float countLetters(string text)
{
    int c, counter = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        c = (int) text[i];
        if ((c >= 65 && c <= 90) || (c >= 97 && c <= 122))
        {
            counter++;
        }
    }
    return counter;
}

// count the number of words in a given text
float countWords(string text)
{
    int counter = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == ' ')
        {
            counter++;
        }
    }
    return counter + 1;
}

// count the number of sentences in a given text
float countSentences(string text)
{
    int counter = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            counter++;
        }
    }
    return counter;
}

// print out the grade for the text given
void readerGrade(float index)
{
    index = round(index);
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 0 && index < 16)
    {
        printf("Grade %i\n", (int) index);
    }
    else
    {
        printf("Grade 16+\n");
    }
}

File: python/readability/readability.py
# get the get_string function from cs50
from cs50 import get_string

# prompt the user for text
text = get_string("Text: ")

# initialize variables
n_letters = n_words = n_sentences = 0

# count letter, words and sentences in text
for c in text:
    # check if is a letter
    if c.isalpha():
        n_letters += 1

    # check if is a space
    elif c.isspace():
        n_words += 1

    # check for end of sentence
    elif c == "!" or c == "." or c == "?":
        n_sentences += 1

# correct number of words
n_words += 1
# if number of senteces has the value of 0
# set the value as 1
n_sentences = n_sentences or 1

# calculate the averages
L = (n_letters / n_words) * 100
S = (n_sentences / n_words) * 100

# use Coleman-Liau index formula
index = round(0.0588 * L - 0.296 * S - 15.8)

# check the reader grade
if index < 1:
    print("Before Grade 1")
elif index > 0 and index < 16:
    print(f"Grade {index}")
else:
    print("Grade 16+")
